// devextreme-vue/index.js
/*!
 * devextreme-vue
 * Version: 22.1.4
 * Build date: Fri Jul 22 2022
 *
 * Copyright (c) 2012 - 2022 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DxRangeSlider = exports.DxRangeSelector = exports.DxRadioGroup = exports.DxProgressBar = exports.DxPopup = exports.DxPopover = exports.DxPolarChart = exports.DxPivotGridFieldChooser = exports.DxPivotGrid = exports.DxPieChart = exports.DxNumberBox = exports.DxNavBar = exports.DxMultiView = exports.DxMenu = exports.DxMap = exports.DxLookup = exports.DxLoadPanel = exports.DxLoadIndicator = exports.DxList = exports.DxLinearGauge = exports.DxHtmlEditor = exports.DxGantt = exports.DxGallery = exports.DxFunnel = exports.DxForm = exports.DxFilterBuilder = exports.DxFileUploader = exports.DxFileManager = exports.DxDropDownButton = exports.DxDropDownBox = exports.DxDrawer = exports.DxDraggable = exports.DxDiagram = exports.DxDeferRendering = exports.DxDateBox = exports.DxDataGrid = exports.DxContextMenu = exports.DxColorBox = exports.DxCircularGauge = exports.DxCheckBox = exports.DxChart = exports.DxCalendar = exports.DxButtonGroup = exports.DxButton = exports.DxBullet = exports.DxBox = exports.DxBarGauge = exports.DxAutocomplete = exports.DxActionSheet = exports.DxAccordion = void 0;
exports.DxVectorMap = exports.DxValidator = exports.DxValidationSummary = exports.DxValidationGroup = exports.DxTreeView = exports.DxTreeMap = exports.DxTreeList = exports.DxTooltip = exports.DxToolbar = exports.DxToast = exports.DxTileView = exports.DxTextBox = exports.DxTextArea = exports.DxTagBox = exports.DxTabs = exports.DxTabPanel = exports.DxSwitch = exports.DxSpeedDialAction = exports.DxSparkline = exports.DxSortable = exports.DxSlider = exports.DxSlideOutView = exports.DxSlideOut = exports.DxSelectBox = exports.DxScrollView = exports.DxScheduler = exports.DxSankey = exports.DxResponsiveBox = exports.DxResizable = exports.DxRecurrenceEditor = void 0;
var accordion_1 = require("./accordion");
Object.defineProperty(exports, "DxAccordion", { enumerable: true, get: function () { return accordion_1.DxAccordion; } });
var action_sheet_1 = require("./action-sheet");
Object.defineProperty(exports, "DxActionSheet", { enumerable: true, get: function () { return action_sheet_1.DxActionSheet; } });
var autocomplete_1 = require("./autocomplete");
Object.defineProperty(exports, "DxAutocomplete", { enumerable: true, get: function () { return autocomplete_1.DxAutocomplete; } });
var bar_gauge_1 = require("./bar-gauge");
Object.defineProperty(exports, "DxBarGauge", { enumerable: true, get: function () { return bar_gauge_1.DxBarGauge; } });
var box_1 = require("./box");
Object.defineProperty(exports, "DxBox", { enumerable: true, get: function () { return box_1.DxBox; } });
var bullet_1 = require("./bullet");
Object.defineProperty(exports, "DxBullet", { enumerable: true, get: function () { return bullet_1.DxBullet; } });
var button_1 = require("./button");
Object.defineProperty(exports, "DxButton", { enumerable: true, get: function () { return button_1.DxButton; } });
var button_group_1 = require("./button-group");
Object.defineProperty(exports, "DxButtonGroup", { enumerable: true, get: function () { return button_group_1.DxButtonGroup; } });
var calendar_1 = require("./calendar");
Object.defineProperty(exports, "DxCalendar", { enumerable: true, get: function () { return calendar_1.DxCalendar; } });
var chart_1 = require("./chart");
Object.defineProperty(exports, "DxChart", { enumerable: true, get: function () { return chart_1.DxChart; } });
var check_box_1 = require("./check-box");
Object.defineProperty(exports, "DxCheckBox", { enumerable: true, get: function () { return check_box_1.DxCheckBox; } });
var circular_gauge_1 = require("./circular-gauge");
Object.defineProperty(exports, "DxCircularGauge", { enumerable: true, get: function () { return circular_gauge_1.DxCircularGauge; } });
var color_box_1 = require("./color-box");
Object.defineProperty(exports, "DxColorBox", { enumerable: true, get: function () { return color_box_1.DxColorBox; } });
var context_menu_1 = require("./context-menu");
Object.defineProperty(exports, "DxContextMenu", { enumerable: true, get: function () { return context_menu_1.DxContextMenu; } });
var data_grid_1 = require("./data-grid");
Object.defineProperty(exports, "DxDataGrid", { enumerable: true, get: function () { return data_grid_1.DxDataGrid; } });
var date_box_1 = require("./date-box");
Object.defineProperty(exports, "DxDateBox", { enumerable: true, get: function () { return date_box_1.DxDateBox; } });
var defer_rendering_1 = require("./defer-rendering");
Object.defineProperty(exports, "DxDeferRendering", { enumerable: true, get: function () { return defer_rendering_1.DxDeferRendering; } });
var diagram_1 = require("./diagram");
Object.defineProperty(exports, "DxDiagram", { enumerable: true, get: function () { return diagram_1.DxDiagram; } });
var draggable_1 = require("./draggable");
Object.defineProperty(exports, "DxDraggable", { enumerable: true, get: function () { return draggable_1.DxDraggable; } });
var drawer_1 = require("./drawer");
Object.defineProperty(exports, "DxDrawer", { enumerable: true, get: function () { return drawer_1.DxDrawer; } });
var drop_down_box_1 = require("./drop-down-box");
Object.defineProperty(exports, "DxDropDownBox", { enumerable: true, get: function () { return drop_down_box_1.DxDropDownBox; } });
var drop_down_button_1 = require("./drop-down-button");
Object.defineProperty(exports, "DxDropDownButton", { enumerable: true, get: function () { return drop_down_button_1.DxDropDownButton; } });
var file_manager_1 = require("./file-manager");
Object.defineProperty(exports, "DxFileManager", { enumerable: true, get: function () { return file_manager_1.DxFileManager; } });
var file_uploader_1 = require("./file-uploader");
Object.defineProperty(exports, "DxFileUploader", { enumerable: true, get: function () { return file_uploader_1.DxFileUploader; } });
var filter_builder_1 = require("./filter-builder");
Object.defineProperty(exports, "DxFilterBuilder", { enumerable: true, get: function () { return filter_builder_1.DxFilterBuilder; } });
var form_1 = require("./form");
Object.defineProperty(exports, "DxForm", { enumerable: true, get: function () { return form_1.DxForm; } });
var funnel_1 = require("./funnel");
Object.defineProperty(exports, "DxFunnel", { enumerable: true, get: function () { return funnel_1.DxFunnel; } });
var gallery_1 = require("./gallery");
Object.defineProperty(exports, "DxGallery", { enumerable: true, get: function () { return gallery_1.DxGallery; } });
var gantt_1 = require("./gantt");
Object.defineProperty(exports, "DxGantt", { enumerable: true, get: function () { return gantt_1.DxGantt; } });
var html_editor_1 = require("./html-editor");
Object.defineProperty(exports, "DxHtmlEditor", { enumerable: true, get: function () { return html_editor_1.DxHtmlEditor; } });
var linear_gauge_1 = require("./linear-gauge");
Object.defineProperty(exports, "DxLinearGauge", { enumerable: true, get: function () { return linear_gauge_1.DxLinearGauge; } });
var list_1 = require("./list");
Object.defineProperty(exports, "DxList", { enumerable: true, get: function () { return list_1.DxList; } });
var load_indicator_1 = require("./load-indicator");
Object.defineProperty(exports, "DxLoadIndicator", { enumerable: true, get: function () { return load_indicator_1.DxLoadIndicator; } });
var load_panel_1 = require("./load-panel");
Object.defineProperty(exports, "DxLoadPanel", { enumerable: true, get: function () { return load_panel_1.DxLoadPanel; } });
var lookup_1 = require("./lookup");
Object.defineProperty(exports, "DxLookup", { enumerable: true, get: function () { return lookup_1.DxLookup; } });
var map_1 = require("./map");
Object.defineProperty(exports, "DxMap", { enumerable: true, get: function () { return map_1.DxMap; } });
var menu_1 = require("./menu");
Object.defineProperty(exports, "DxMenu", { enumerable: true, get: function () { return menu_1.DxMenu; } });
var multi_view_1 = require("./multi-view");
Object.defineProperty(exports, "DxMultiView", { enumerable: true, get: function () { return multi_view_1.DxMultiView; } });
var nav_bar_1 = require("./nav-bar");
Object.defineProperty(exports, "DxNavBar", { enumerable: true, get: function () { return nav_bar_1.DxNavBar; } });
var number_box_1 = require("./number-box");
Object.defineProperty(exports, "DxNumberBox", { enumerable: true, get: function () { return number_box_1.DxNumberBox; } });
var pie_chart_1 = require("./pie-chart");
Object.defineProperty(exports, "DxPieChart", { enumerable: true, get: function () { return pie_chart_1.DxPieChart; } });
var pivot_grid_1 = require("./pivot-grid");
Object.defineProperty(exports, "DxPivotGrid", { enumerable: true, get: function () { return pivot_grid_1.DxPivotGrid; } });
var pivot_grid_field_chooser_1 = require("./pivot-grid-field-chooser");
Object.defineProperty(exports, "DxPivotGridFieldChooser", { enumerable: true, get: function () { return pivot_grid_field_chooser_1.DxPivotGridFieldChooser; } });
var polar_chart_1 = require("./polar-chart");
Object.defineProperty(exports, "DxPolarChart", { enumerable: true, get: function () { return polar_chart_1.DxPolarChart; } });
var popover_1 = require("./popover");
Object.defineProperty(exports, "DxPopover", { enumerable: true, get: function () { return popover_1.DxPopover; } });
var popup_1 = require("./popup");
Object.defineProperty(exports, "DxPopup", { enumerable: true, get: function () { return popup_1.DxPopup; } });
var progress_bar_1 = require("./progress-bar");
Object.defineProperty(exports, "DxProgressBar", { enumerable: true, get: function () { return progress_bar_1.DxProgressBar; } });
var radio_group_1 = require("./radio-group");
Object.defineProperty(exports, "DxRadioGroup", { enumerable: true, get: function () { return radio_group_1.DxRadioGroup; } });
var range_selector_1 = require("./range-selector");
Object.defineProperty(exports, "DxRangeSelector", { enumerable: true, get: function () { return range_selector_1.DxRangeSelector; } });
var range_slider_1 = require("./range-slider");
Object.defineProperty(exports, "DxRangeSlider", { enumerable: true, get: function () { return range_slider_1.DxRangeSlider; } });
var recurrence_editor_1 = require("./recurrence-editor");
Object.defineProperty(exports, "DxRecurrenceEditor", { enumerable: true, get: function () { return recurrence_editor_1.DxRecurrenceEditor; } });
var resizable_1 = require("./resizable");
Object.defineProperty(exports, "DxResizable", { enumerable: true, get: function () { return resizable_1.DxResizable; } });
var responsive_box_1 = require("./responsive-box");
Object.defineProperty(exports, "DxResponsiveBox", { enumerable: true, get: function () { return responsive_box_1.DxResponsiveBox; } });
var sankey_1 = require("./sankey");
Object.defineProperty(exports, "DxSankey", { enumerable: true, get: function () { return sankey_1.DxSankey; } });
var scheduler_1 = require("./scheduler");
Object.defineProperty(exports, "DxScheduler", { enumerable: true, get: function () { return scheduler_1.DxScheduler; } });
var scroll_view_1 = require("./scroll-view");
Object.defineProperty(exports, "DxScrollView", { enumerable: true, get: function () { return scroll_view_1.DxScrollView; } });
var select_box_1 = require("./select-box");
Object.defineProperty(exports, "DxSelectBox", { enumerable: true, get: function () { return select_box_1.DxSelectBox; } });
var slide_out_1 = require("./slide-out");
Object.defineProperty(exports, "DxSlideOut", { enumerable: true, get: function () { return slide_out_1.DxSlideOut; } });
var slide_out_view_1 = require("./slide-out-view");
Object.defineProperty(exports, "DxSlideOutView", { enumerable: true, get: function () { return slide_out_view_1.DxSlideOutView; } });
var slider_1 = require("./slider");
Object.defineProperty(exports, "DxSlider", { enumerable: true, get: function () { return slider_1.DxSlider; } });
var sortable_1 = require("./sortable");
Object.defineProperty(exports, "DxSortable", { enumerable: true, get: function () { return sortable_1.DxSortable; } });
var sparkline_1 = require("./sparkline");
Object.defineProperty(exports, "DxSparkline", { enumerable: true, get: function () { return sparkline_1.DxSparkline; } });
var speed_dial_action_1 = require("./speed-dial-action");
Object.defineProperty(exports, "DxSpeedDialAction", { enumerable: true, get: function () { return speed_dial_action_1.DxSpeedDialAction; } });
var switch_1 = require("./switch");
Object.defineProperty(exports, "DxSwitch", { enumerable: true, get: function () { return switch_1.DxSwitch; } });
var tab_panel_1 = require("./tab-panel");
Object.defineProperty(exports, "DxTabPanel", { enumerable: true, get: function () { return tab_panel_1.DxTabPanel; } });
var tabs_1 = require("./tabs");
Object.defineProperty(exports, "DxTabs", { enumerable: true, get: function () { return tabs_1.DxTabs; } });
var tag_box_1 = require("./tag-box");
Object.defineProperty(exports, "DxTagBox", { enumerable: true, get: function () { return tag_box_1.DxTagBox; } });
var text_area_1 = require("./text-area");
Object.defineProperty(exports, "DxTextArea", { enumerable: true, get: function () { return text_area_1.DxTextArea; } });
var text_box_1 = require("./text-box");
Object.defineProperty(exports, "DxTextBox", { enumerable: true, get: function () { return text_box_1.DxTextBox; } });
var tile_view_1 = require("./tile-view");
Object.defineProperty(exports, "DxTileView", { enumerable: true, get: function () { return tile_view_1.DxTileView; } });
var toast_1 = require("./toast");
Object.defineProperty(exports, "DxToast", { enumerable: true, get: function () { return toast_1.DxToast; } });
var toolbar_1 = require("./toolbar");
Object.defineProperty(exports, "DxToolbar", { enumerable: true, get: function () { return toolbar_1.DxToolbar; } });
var tooltip_1 = require("./tooltip");
Object.defineProperty(exports, "DxTooltip", { enumerable: true, get: function () { return tooltip_1.DxTooltip; } });
var tree_list_1 = require("./tree-list");
Object.defineProperty(exports, "DxTreeList", { enumerable: true, get: function () { return tree_list_1.DxTreeList; } });
var tree_map_1 = require("./tree-map");
Object.defineProperty(exports, "DxTreeMap", { enumerable: true, get: function () { return tree_map_1.DxTreeMap; } });
var tree_view_1 = require("./tree-view");
Object.defineProperty(exports, "DxTreeView", { enumerable: true, get: function () { return tree_view_1.DxTreeView; } });
var validation_group_1 = require("./validation-group");
Object.defineProperty(exports, "DxValidationGroup", { enumerable: true, get: function () { return validation_group_1.DxValidationGroup; } });
var validation_summary_1 = require("./validation-summary");
Object.defineProperty(exports, "DxValidationSummary", { enumerable: true, get: function () { return validation_summary_1.DxValidationSummary; } });
var validator_1 = require("./validator");
Object.defineProperty(exports, "DxValidator", { enumerable: true, get: function () { return validator_1.DxValidator; } });
var vector_map_1 = require("./vector-map");
Object.defineProperty(exports, "DxVectorMap", { enumerable: true, get: function () { return vector_map_1.DxVectorMap; } });
