// devextreme-vue/slide-out-view.js
/*!
 * devextreme-vue
 * Version: 22.1.4
 * Build date: Fri Jul 22 2022
 *
 * Copyright (c) 2012 - 2022 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DxSlideOutView = void 0;
var slide_out_view_1 = __importDefault(require("devextreme/ui/slide_out_view"));
var index_1 = require("./core/index");
var DxSlideOutView = index_1.createComponent({
    props: {
        activeStateEnabled: Boolean,
        contentTemplate: {},
        disabled: Boolean,
        elementAttr: Object,
        height: [Function, Number, String],
        hint: String,
        hoverStateEnabled: Boolean,
        menuPosition: String,
        menuTemplate: {},
        menuVisible: Boolean,
        onDisposing: Function,
        onInitialized: Function,
        onOptionChanged: Function,
        rtlEnabled: Boolean,
        swipeEnabled: Boolean,
        visible: Boolean,
        width: [Function, Number, String]
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:activeStateEnabled": null,
        "update:contentTemplate": null,
        "update:disabled": null,
        "update:elementAttr": null,
        "update:height": null,
        "update:hint": null,
        "update:hoverStateEnabled": null,
        "update:menuPosition": null,
        "update:menuTemplate": null,
        "update:menuVisible": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onOptionChanged": null,
        "update:rtlEnabled": null,
        "update:swipeEnabled": null,
        "update:visible": null,
        "update:width": null,
    },
    computed: {
        instance: function () {
            return this.$_instance;
        }
    },
    beforeCreate: function () {
        this.$_WidgetClass = slide_out_view_1.default;
        this.$_hasAsyncTemplate = true;
    }
});
exports.DxSlideOutView = DxSlideOutView;
exports.default = DxSlideOutView;
