// v-title/lib/directives/title.js
/*!
 * v-title v2.0.1
 * (c) 2019-2022 yujinpan
 * Released under the MIT License.
 */

import { toSize, toNum, isOverflow } from '../utils/dom.js';
import { TitleStore } from '../utils/store.js';
import tippy from 'tippy.js';
import 'tippy.js/dist/tippy.css';
import 'tippy.js/themes/light-border.css';

/**
 * 辅助文字展示（类似el-tooltip）
 *
 * @modifier light 主题 translucent/light，默认 translucent
 * @modifier overflow 是否溢出模式，文字显示省略号时才显示，默认 false
 * @modifier multiple 是否多行模式，文字多行显示省略号时才显示，默认 false
 * @modifier delay 是否多行模式，文字多行显示省略号时才显示，默认 false
 * @attr {Number} title-delay-time 延迟时间，默认 0
 * @attr {String} title-placement 位置 top(-start, -end), right(-start, -end), bottom(-start, -end), left(-start, -end)，默认 top
 * @attr {String} title-max-width 最大宽度
 * @example
 * ```
 * <div v-title.overflow.translucent="'test'" title-placement="top"></div>
 * ```
 */
var title = {
  inserted: function inserted(el, binding) {
    var theme = binding.modifiers.light ? 'light-border' : 'translucent';
    var placement = el.getAttribute('title-placement') || 'top';
    var trigger = el.getAttribute('title-trigger') || undefined;
    var maxWidth = toSize(el.getAttribute('title-max-width'));
    var delayTime = toNum(el.getAttribute('title-delay-time')) || 200;
    var overflow = binding.modifiers.overflow;
    var multiple = binding.modifiers.multiple;
    var delay = binding.modifiers.delay;
    var instance = tippy(el, {
      content: binding.value,
      theme: theme,
      trigger: trigger,
      maxWidth: maxWidth,
      placement: placement,
      delay: !delay ? 0 : [delayTime],
      interactive: true,
      appendTo: document.body,
      onShow: function onShow(instance) {
        if (!instance.props.content || overflow && !isOverflow(el, multiple)) return false;
      },
      popperOptions: {
        modifiers: {
          preventOverflow: {
            enabled: false
          },
          hide: {
            enabled: false
          }
        },
        removeOnDestroy: true
      }
    });
    TitleStore.add(el, instance);
  },
  componentUpdated: function componentUpdated(el, binding) {
    var instance = TitleStore.get(el);
    instance && instance.setContent(binding.value);
  },
  unbind: function unbind(el) {
    var instance = TitleStore.get(el);

    if (instance) {
      instance.destroy();
      TitleStore.remove(el);
    }
  }
};

export default title;
