// devextreme-vue/core/strategy/vue2/extension-component.js
/*!
 * devextreme-vue
 * Version: 22.1.4
 * Build date: Fri Jul 22 2022
 *
 * Copyright (c) 2012 - 2022 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DxExtensionComponent = void 0;
var component_1 = require("./component");
var DxExtensionComponent = function () { return component_1.BaseComponent().extend({
    created: function () {
        this.$_isExtension = true;
    },
    mounted: function () {
        this.$el.setAttribute("isExtension", "true");
        if (this.$vnode && this.$vnode.componentOptions.$_hasOwner) {
            return;
        }
        this.attachTo(this.$el);
    },
    methods: {
        attachTo: function (element) {
            this.$_createWidget(element);
        }
    }
}); };
exports.DxExtensionComponent = DxExtensionComponent;
